### ISSUE TITLE
FrameworkReference 'Microsoft.AspNetCore.App' was not recognized after downgrade

### ISSUE THREAD

**Issue by syndicatedshannon** [dotnet/sdk#3760]
Downgrade TargetFramework from netcoreapp3.0 to netcoreapp2.2 causes the following upon build:

> The FrameworkReference 'Microsoft.AspNetCore.App' was not recognized

Clean/Rebuild All does not resolve. Deleting bin/obj folders in the specific projects reporting the FrameworkReference error does resolve. This causes unexpected failures when switching branches. Note the project also depends upon .NET Standard 2.1/2.0 projects, which have coincident upgrade/downgrades. However, cleaning those projects did not resolve.

I can repro and try deleting something specific if it will help pin the problem down further.

**Comment by RobBowman** [comment]
I'm seeing the same problem but deleting the bin/obj folders has not helped.

**Comment by syndicatedshannon** [comment]
@RobBowman Just to clarify, in my scenario deleting the depended projects output bin/obj did not work, deleting the dependent did.

**Comment by syndicatedshannon** [comment]
@RobBowman In my case, I had completely reverted my project to a prior building version and still received the error, until I manually cleaned intermediate and output. However, I can't recall offhand what references changed between the versions. Is it possible you have two issues, one being the issue I ran into, and another being something you didn't revert?

**Comment by RobBowman** [comment]
@syndicatedshannon  please see: https://stackoverflow.com/questions/58509531/downgrade-from-core-3-to-2-then-microsoft-aspnetcore-app-not-recognised#58509560 . For 2.2 needs to be <PackageReference Include="Microsoft.AspNetCore.App" /> rather than FrameworkReference

**Comment by syndicatedshannon** [comment]
Yeah, that's the kind of thing I meant. That won't have happened in my case where I reverted, unless there was something left behind that clean didn't handle.
